DataFormats/L1TMuon/interface/EMTFOutput.h

```diff
@@ -50,27 +50,27 @@ namespace l1t {
     void set_AMC13Trailer(emtf::AMC13Trailer bits)  { AMC13Trailer = bits; hasAMC13Trailer = true; };
     void set_dataword(uint64_t bits)                { dataword = bits;               };
     
-    bool HasAMC13Header()                   { return hasAMC13Header;  };
-    bool HasMTF7Header()                    { return hasMTF7Header;   };
-    bool HasEventHeader()                   { return hasEventHeader;  };
-    bool HasCounters()                      { return hasCounters;     };
-    int  NumSP()                            { return numSP;           };
-    int  NumRPC()                           { return numRPC;          };
-    int  NumME()                            { return numME;           };
-    bool HasAMC13Trailer()                  { return hasAMC13Trailer; };
-    bool HasMTF7Trailer()                   { return hasMTF7Trailer;  };
-    bool HasEventTrailer()                  { return hasEventTrailer; };
-    emtf::AMC13Header GetAMC13Header()      { return AMC13Header;   };
-    emtf::MTF7Header GetMTF7Header()        { return MTF7Header;    };
-    emtf::EventHeader GetEventHeader()      { return EventHeader;   };
-    emtf::Counters GetCounters()            { return Counters;      };
-    emtf::MECollection GetMECollection()    { return MECollection;  };
-    emtf::RPCCollection GetRPCCollection()  { return RPCCollection; };
-    emtf::SPCollection GetSPCollection()    { return SPCollection;  };
-    emtf::EventTrailer GetEventTrailer()    { return EventTrailer;  };
-    emtf::MTF7Trailer GetMTF7Trailer()      { return MTF7Trailer;   };
-    emtf::AMC13Trailer GetAMC13Trailer()    { return AMC13Trailer;  };
-    const uint64_t Dataword() const     { return dataword; };
+    const bool HasAMC13Header()                  const { return hasAMC13Header;  };
```

Comment:
i understand that the methods should be constant - but not their return types - what were the errors?

> On Mar 17, 2016, at 5:37 PM, abrinke1 <notifications@github.com> wrote:
> 
> In DataFormats/L1TMuon/interface/EMTFOutput.h:
> 
> > -    int  NumME()                            { return numME;           };
> > -    bool HasAMC13Trailer()                  { return hasAMC13Trailer; };
> > -    bool HasMTF7Trailer()                   { return hasMTF7Trailer;  };
> > -    bool HasEventTrailer()                  { return hasEventTrailer; };
> > -    emtf::AMC13Header GetAMC13Header()      { return AMC13Header;   };
> > -    emtf::MTF7Header GetMTF7Header()        { return MTF7Header;    };
> > -    emtf::EventHeader GetEventHeader()      { return EventHeader;   };
> > -    emtf::Counters GetCounters()            { return Counters;      };
> > -    emtf::MECollection GetMECollection()    { return MECollection;  };
> > -    emtf::RPCCollection GetRPCCollection()  { return RPCCollection; };
> > -    emtf::SPCollection GetSPCollection()    { return SPCollection;  };
> > -    emtf::EventTrailer GetEventTrailer()    { return EventTrailer;  };
> > -    emtf::MTF7Trailer GetMTF7Trailer()      { return MTF7Trailer;   };
> > -    emtf::AMC13Trailer GetAMC13Trailer()    { return AMC13Trailer;  };
> > -    const uint64_t Dataword() const     { return dataword; };
> > +    const bool HasAMC13Header()                  const { return hasAMC13Header;  };
> 
> Not sure why they need to be const, but without it there were compile errors, and with it, it works.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
>